crates/types/src/light_client.rs

```diff
@@ -174,57 +207,51 @@ impl<F: PrimeField> From<Vec<F>> for GenericPublicInput<F> {
 }
 
 impl<F: PrimeField> GenericPublicInput<F> {
-    /// Return the threshold
-    #[must_use]
-    pub fn threshold(&self) -> F {
-        self.0[0]
-    }
-
     /// Return the view number of the light client state
     #[must_use]
     pub fn view_number(&self) -> F {
-        self.0[1]
+        self.0[0]
     }
 
     /// Return the block height of the light client state
     #[must_use]
     pub fn block_height(&self) -> F {
-        self.0[2]
+        self.0[1]
     }
 
     /// Return the block commitment root of the light client state
     #[must_use]
     pub fn block_comm_root(&self) -> F {
-        self.0[3]
-    }
-
-    /// Return the fee ledger commitment of the light client state
-    #[must_use]
-    pub fn fee_ledger_comm(&self) -> F {
-        self.0[4]
+        self.0[2]
     }
 
     /// Return the stake table commitment of the light client state
     #[must_use]
     pub fn stake_table_comm(&self) -> (F, F, F) {
-        (self.0[5], self.0[6], self.0[7])
+        (self.0[3], self.0[4], self.0[5])
     }
 
     /// Return the qc key commitment of the light client state
     #[must_use]
     pub fn qc_key_comm(&self) -> F {
-        self.0[5]
+        self.0[3]
     }
 
     /// Return the state key commitment of the light client state
     #[must_use]
     pub fn state_key_comm(&self) -> F {
-        self.0[6]
+        self.0[4]
     }
 
     /// Return the stake amount commitment of the light client state
     #[must_use]
     pub fn stake_amount_comm(&self) -> F {
-        self.0[7]
+        self.0[5]
+    }
+
+    /// Return the threshold
+    #[must_use]
+    pub fn threshold(&self) -> F {
+        self.0[6]
```

Comment:
All of these magic numbers are kind of tricky to reason about, can we just use a named  struct?

Reply:
@alxiong @alysiahuggins do you remember if there is any particular reason why we don't use a named struct?

Reply:
this is the decision made by @mrain and i recall the public input right now is simple enough to keep track of? 
i also personally prefer a named struct, less error prone. 

but i tried to make minimal changes for this task, so kept the legacy way. Can update in the future as tech debt.